src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,154 @@
+package parser;
+
+import java.lang.reflect.Array;
+import java.util.ArrayList;
+
+import seedu.command.*;
+
+import java.util.Collections;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parse user input
+ * Referenced https://github.com/se-edu/addressbook-level2/blob/master/src/seedu/addressbook/parser/Parser.java
+ */
+public class Parser {
+
+    /**
+     * Format to parse command by breaking it up into two segments: command and arguments: these can then be separately
+     * passed into arguments.
+     */
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)\\s+(?<arguments>.*)");
+    public static final Pattern ADD_COMMAND_FORMAT = Pattern.compile(
+            "n/(?<itemName>.+)" + "\\s+" +
+                    "sn/(?<serialNumber>.+)" + "\\s+" +
+                    "t/(?<equipmentType>.+)" + "\\s+" +
+                    "c/\\$?(?<cost>[\\d,\\.]+)" + "\\s+" + // TODO: maybe have error checking formula separately?
```

Comment:
I think we should do the error checking separately. Later on when we actually try to add the equipment in the command execution stage